cores/esp8266/i2s.h

```diff
@@ -51,6 +51,7 @@ bool i2s_write_lr(int16_t left, int16_t right);//combines both channels and call
 bool i2s_is_full();//returns true if DMA is full and can not take more bytes (overflow)
 bool i2s_is_empty();//returns true if DMA is empty (underflow)
 int16_t i2s_available();// returns the number of samples than can be written before blocking
+void i2s_set_callback(void (*callback) (void));
```

Comment:
(continuing prior thought)
The actual callback function passed in, though, needs to be in IRAM.  A comment in the code to that effect would be appreciated here, so it's obvious to people coming into it that it's an ISR and needs to be very special...